doc/migration.md

```diff
@@ -1,46 +1,66 @@
-indicate that it wants to speak the `p.haul` protocol, instead of just using `rsync`
-to transfer the images over slowly).
-
-The sink then examines this message and responds with whatever it
-supports. Continuing our example, if the sink is not on a Btrfs
-file system, it responds with the lowest common denominator (`rsync`, in
-this case), and the source is to send the root file system using `rsync`.
-Similarly with the CRIU connection; if the sink doesn't have support for
-the `p.haul` protocol (or whatever), we fall back to `rsync`.
+# How to migrate existing LXD instances between servers
+
+To move an instance from one LXD server to another, use the `lxc move` command:
+
+    lxc move [<source_remote>:]<source_instance_name> <target_remote>:[<target_instance_name>]
+
+```{note}
+When moving a container, you must stop it first.
+See {ref}`live-migration` for more information.
+```
+
+You don't need to specify the source remote if it is your default remote, and you can leave out the target instance name if you want to use the same instance name.
+If you want to move the instance to a specific cluster member, specify it with the `--target` flag.
+
+You can add the `--mode` flag to choose a transfer mode, depending on your network setup:
+
+`pull` (default)
+: Instruct the target server to pull the respective instance.
+
+`push`
+: Push the instance from the source server to the target server.
+
+`relay`
+: Pull the instance from the source server to the local client, and then push it to the target server.
+
+If you need to adapt the configuration for the instance to run on the target server, you can either specify the new configuration directly (using `--config`, `--device`, `--storage` or `--target-project`) or through profiles (using `--no-profiles` or `--profile`). See `lxc move --help` for all available flags.
+
+(live-migration)=
+## Live migration
+
+Virtual machines can be moved to another server while they are running, thus without any downtime.
+
+For containers, there is limited support for live migration using [{abbr}`CRIU (Checkpoint/Restore in Userspace)`](https://criu.org/).
+However, because of extensive kernel dependencies, only very basic containers (non-`systemd` containers without a network device) can be migrated reliably.
+In most real-world scenarios, you should stop the container, move it over and then start it again.
+
+If you want to use live migration for containers, you must enable CRIU on both the source and the target server.
+If you are using the snap, use the following commands to enable CRIU:
+
+    snap set lxd criu.enable=true
+    systemctl reload snap.lxd.daemon
+
+### CRIU migration
+
+CRIU migration uses three WebSockets (channels):
+
+1. The control stream
+1. The CRIU images stream
+1. The file-system stream
+
+The protocol used on the CRIU channel and the file-system channel varies depending on what is negotiated over the control socket (see below).
+
+The migration happens in the following way:
```

Comment:
This is purely based on the existing documentation. Is it still correct?

Reply:
It's vaguely correct. It ignores virtual machines and ignores the fact that we have push and relay migration too now.

In general, that's the section I was telling you is far too low level to make any sense in our docs and I think should just be plain removed. At some point we may want to start building up internal development documentation and so re-introduce something like this, but it will need a full rewrite anyway.

Reply:
OK, let's get rid of it then. :)